### ISSUE TITLE
/32 mask opens access to any host

### ISSUE THREAD

**Issue by mruncles** [yandex/ClickHouse#3770]
ClickHouse server version 18.14.18 revision 54409 (18.12.17 same thing)

configuration like this
`            <networks incl="networks" replace="replace">
                <ip>::1</ip>
                <ip>127.0.0.1</ip>
                <ip>172.24.35.72/32</ip>
            </networks>
`
allows access from, for example, 172.24.41.201

If I remove '/32' from configuration, network acl starts to work as intended.

If I recall correctly, I did not observe such behavior before upgrading to 18.12.17 from some 1.1.54xx version.

**Comment by filimonov** [comment]
Confirm. The problem appears for **any** netmask in CIDR format. 
Test case:
```xml
<?xml version="1.0"?>
<yandex>
    <users>
        <readonly>
            <password></password>
            <networks incl="networks" replace="replace">
                <ip>1.0.0.1/24</ip>
            </networks>
            <profile>readonly</profile>
            <quota>default</quota>
        </readonly>
    </users>
</yandex>
```

With that configuration access from localhost should be forbidden. But 
```clickhouse-client -u readonly --query="SELECT 'ups'"```

When changing ip to '1.0.0.1/255.255.255.0' it works as expected.

**Comment by iameugenejo** [comment]
I'm having the same issue with 18.5.1

**Comment by liangliangxiao** [comment]
I'm having the same issue with ClickHouse server version 18.16.1 revision 54412.

**Comment by ztlpn** [comment]
This error was fixed in #3945. The fix is available in 19.1.6.
